src/feature/history/ui/history.uielement.ts

```diff
@@ -0,0 +1,264 @@
+import {
+    Button,
+    Col,
+    collection,
+    Container,
+    Datatable,
+    documents,
+    Form,
+    formatCurrency,
+    formatDatetime,
+    formatToken,
+    Fragment,
+    GridTile,
+    Input,
+    isBusy,
+    PageHeaderTile,
+    Row,
+    Select,
+    Span,
+    UiElement,
+  } from '@earnkeeper/ekp-sdk';
+  import { DEFAULT_HISTORY_FORM } from '../../../util';
+  import { HistoryDocument } from './history.document';
+  
+  export default function element(): UiElement {
+    return Container({
+      children: [
+        Row({
+          className: 'mb-2',
+          children: [
+            Col({
+              className: 'col-auto',
+              children: [
+                PageHeaderTile({
+                  title: 'Player History',
+                  icon: 'cid-history',
+                }),
+              ],
+            }),
+          ],
+        }),
+        Span({
+          className: 'd-block mt-1 mb-2 font-small-3',
+          content:
+            'The official splinterlands player history, select player to update.',
+        }),
+        formRow(),
+        historyRow(),
+      ],
+    });
+  }
+  
+  function formRow(): UiElement {
+    return Form({
+      name: 'history',
+      schema: {
+        type: 'object',
+        properties: {
+          playername: 'string',
+        },
+        default: DEFAULT_HISTORY_FORM,
+      },
+      children: [
+        Row({
+          className: 'mb-1',
+          children: [
+            Col({
+              className: 'col-12 col-md-auto',
+              children: [
+                Input({
+                  label: 'Player',
+                  name: 'playername',
+                }),
+              ],
+            }),
+             Col({
+              className: 'col-12 col-md-auto my-auto',
+              children: [
+                Button({
+                  label: 'View',
+                  isSubmit: true,
+                  busyWhen: isBusy(collection(HistoryDocument)),
+                }),
+              ],
+            }),
+          ],
+        }),
+      ],
+    });
+  }
+  
+function historyRow(): UiElement {
+  return Datatable({
+    defaultSortFieldId: 'created_date',
+    defaultSortAsc: true,
+    defaultView: {
+      xs: 'grid',
+      lg: 'column',
+    },
+    data: documents(HistoryDocument),
+    busyWhen: isBusy(collection(HistoryDocument)),
+    filters: [
+      {
+        columnId: 'mana_cap',
+        type: 'slider',
+      },
+      
+      {
+        columnId: 'rshares',
+        type: 'slider',
+      },
+      
+    ],
+
+    gridView: {
+      tileWidth: [12, 6, 4, 3],
+      tile: GridTile({
+        image: Fragment(),
+        details: [
+          {
+            label: 'Date',
+            value: '$.created_date',
```

Comment:
Remember to change these values too when changing HistoryDocument to lowerCamelCase

Reply:
done